src/riak_cs_s3_auth.erl

```diff
@@ -43,20 +44,38 @@
 
 -spec identify(RD::term(), #context{}) ->
                       {string() | undefined,
-                       string() | {v4, v4_attrs()} | undefined}.
+                       string() | {v4, v4_attrs()} | undefined} |
+                      {failed, Reason::atom()}.
```

Comment:
`parse_auth_header/1` may return `{failed, ...}` although it's the second element is not atom.